lib/concurrent/utility/monotonic_time.rb

```diff
@@ -0,0 +1,147 @@
+module Concurrent
+
+  # Clock that cannot be set and represents monotonic time since
+  # some unspecified starting point.
+  # @!visibility private
+  GLOBAL_MONOTONIC_CLOCK = Class.new {
+
+    if defined?(Process::CLOCK_MONOTONIC)
+      # @!visibility private
+      def get_time
+        Process.clock_gettime(Process::CLOCK_MONOTONIC)
+      end
+    elsif RUBY_PLATFORM == 'java'
+      # @!visibility private
+      def get_time
+        java.lang.System.nanoTime() / 1_000_000_000.0
+      end
+    else
+
+      require 'thread'
+
+      # @!visibility private
+      def initialize
+        @mutex = Mutex.new
+        @correction = 0
+        @last_time = Time.now.to_f
+      end
+
+      # @!visibility private
+      def get_time
+        @mutex.synchronize {
+          @correction ||= 0 # compensating any back time shifts
+          now = Time.now.to_f
+          corrected_now = now + @correction
+          if @last_time < corrected_now
+            @last_time = corrected_now 
+          else
+            @correction += @last_time - corrected_now + 0.000_001
+            @last_time = @correction + now
+          end
+        }
+      end
+    end
+  }.new
```

Comment:
I see your point. Nevertheless It already happen to me more than once that I though there is no use-case for something, but there was. For that reason I think it's better to assume that the use-case is just not obvious. In this case testing might be the reason to provide the class, to be able to create new instance injecting mocked Time, then messing with the clock testing that the monotonic condition holds, without interfering with the global instance in the gem.